build_tools/new_library.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+ask() {
+	[ -z "${1+x}" ] && exit 1
+	if [ -z "${2+x}" ]; then
+		read -p "${1}: " rc
+	else
+		rc="$2"
+	fi
+	echo "$rc"
+}
+
+checkpkg() {
+	if [ -f $1.mk ]; then
+		>&2 echo "$1.mk already exists."
+		exit 1
+	elif grep -R "^$pkg:" *.mk &> /dev/null; then
+		>&2 echo "$1 already exists."
+		exit 1
+	fi
+}
+
+checkbuild() {
+	if [ ! -f build_misc/templates/$1-lib.mk ]; then
+		>&2 echo "$1 is not a valid buildsystem"
+		exit 1
+	fi
+}
+
+downloadlink() {
+	if [ "$(${SED} 's|^.*://||' <<< "${1}" | cut -d'/' -f1)" = "github.com" ]; then
+		echo -e "\t$\(call GITHUB_ARCHIVE,$(${SED} 's|^.*://||' <<< "${1}" | cut -d'/' -f2),$(${SED} 's|^.*://||' <<< "${1}" | cut -d'/' -f3),$\(${4}_VERSION\),$(echo "${1}" | rev | cut -d'/' -f1 | rev | ${SED} 's/\.tar.*//g' | ${SED} "s/${2}//g")$\(${4}_VERSION\)\)"
+	else
+		if echo "${1}" | rev | cut -d'/' -f1 | rev | ${SED} 's/\.tar.*//g' | ${SED} "s/${2}//g" | grep "${3}" &>/dev/null; then
+			echo -e "\twget -q -nc -P\$(BUILD_SOURCE) $(${SED} "s/${2}/\$(${4}_VERSION)/g" <<< "$1")"
+		else
+			echo -e "\t-[ ! -f "$\(BUILD_SOURCE\)/${3}-$\(${4}_VERSION\).tar.$(rev <<< "$download" | cut -d'.' -f1 | rev)" ] \&\& \\
+\t	\twget -q -nc -O\$(BUILD_SOURCE)/${3}-\$(${4}_VERSION).tar.$(rev <<< "$download" | cut -d'.' -f1 | rev) \\
+\t	\t\t$(${SED} "s/${2}/\$(${4}_VERSION)/g" <<< "$1")"
+		fi
+	fi
+}
+
+ask_extended_description() {
+if [ -z "$1" ]; then
+	ESCAPED_DESCRIPTION=""
+	>&2 echo "Extended Description (press ENTER then Ctrl+D when done):"
+	while read line
+        	do ESCAPED_DESCRIPTION="$ESCAPED_DESCRIPTION"" $line\n"
+	done
+else
+	ESCAPED_DESCRIPTION="$(echo "$1" | ${SED} '$!s/$/\\n/' | tr -d '\n' | ${SED} 's|^| |g')"
+	fi
+	echo "$ESCAPED_DESCRIPTION"
+}
+createfromtemplate() {
+	${SED} -e "s/@pkg@/${pkg}/g" \
+		-e "s/@PKG@/${formatpkg}/g" \
+		-e "s/@PKG_VERSION@/${ver}/g" \
+		-e "s/@SOVER@/${sover}/g" \
+		-e "s|@DEB_DESCRIPTION@|${description}|g" \
+		-e "s|@DEB_EXTENDED_DESCRIPTION@|${extended_description}|g" \
+		-e "s|@download@|$(downloadlink "$download" "$ver" "$pkg" "$formatpkg")|g" \
+		-e "s|@compression@|$(rev <<< "$download" | cut -d'.' -f1 | rev)|g" \
+		"$1" > "$2"
+}
+
+main() {
+	pkg="$(ask "Package Name" $1)"
+	checkpkg "$pkg"
+	formatpkg="$(${SED} -e 's|-|_|g' -e "s|\(.\)|\u\1|g" <<< "${pkg}")"
+	build="$(ask "Build System" $2)"
+	checkbuild "$build"
+	ver="$(ask "Package Version" $3)"
+	download="$(ask "Download Link" $4)"
+	sover="$(ask "SO Version" $5)"
+	description="$(ask "Description" "$6")"
+	extended_description="$(ask_extended_description "$7")"
+	echo "Writing out control and makefile"
+	createfromtemplate build_misc/templates/${build}-lib.mk ${pkg}.mk
+	createfromtemplate build_misc/templates/libpkg1.control build_info/${pkg}${sover}.control
```

Comment:
Add the control file to `new_package.sh` and create one file that is sourced with the shared functions so that the two scripts are unified.